servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/SourceToFuture.java

```diff
@@ -122,10 +125,10 @@ private T reportGet(@Nullable final Object value) throws ExecutionException {
         if (value instanceof Throwable) {
             throw new ExecutionException((Throwable) value);
         }
-        if (value == CANCELLED) {
-            throw new CancellationException();
+        if (CancellationWrapper.isCancellationWrapper(value)) {
+            throw ((CancellationWrapper) value).exception;
```

Comment:
As a user, I will much more appreciate if the stack-trace shows me what code path of my business logic throws `CancellationException` (where I blocked) instead of seeing what code path cancelled, because it helps me to understand what request failed. 
The pre-existing approach aligns with Java behavior for `java.util.concurrent.FutureTask`.

If there is a real need to preserve a stack trace that cancelled, consider adding it as a cause or a suppressed exception for the one we create at this line today.

Reply:
[CompletableFuture](https://github.com/openjdk/jdk17/blob/master/src/java.base/share/classes/java/util/concurrent/CompletableFuture.java#L2479) takes the stack trace from "where it was cancelled". I think this provides more value bcz the place that does a `get()` knows its stack, but doesn't know where/why it was cancelled.

As you say, `java.util.concurrent.FutureTask` does it the other way (throws from `get()`).

Reply:
CompletableFuture javadocs talk to this briefly:

https://github.com/openjdk/jdk17/blob/master/src/java.base/share/classes/java/util/concurrent/CompletableFuture.java#L97-L103
```java
 * <p>CompletableFuture also implements {@link Future} with the following
 * policies: <ul>
 *
 * <li>Since (unlike {@link FutureTask}) this class has no direct
 * control over the computation that causes it to be completed,
 * cancellation is treated as just another form of exceptional
 * completion.  Method {@link #cancel cancel} has the same effect as
 * {@code completeExceptionally(new CancellationException())}. Method
 * {@link #isCompletedExceptionally} can be used to determine if a
 * CompletableFuture completed in any exceptional fashion.
```

Reply:
Yeah, no objections on preserving both. It's useful addition. Just based on my experience, these "async exception stack-traces" bring more confusion vs help. So, preserving the caller's stack-trace at the first place is more important. Thanks for improving it!